master/internal/populate_metrics.go

```diff
@@ -0,0 +1,305 @@
+
+	return nil
+}
+
+func reportTrivialMetrics(ctx context.Context, api *apiServer, trialID int32, batches int) error {
+	fmt.Println("trivial metrics for", batches, "batches")
+
+	trainingMetrics := trialv1.TrialMetrics{
+		TrialId:        trialID,
+		StepsCompleted: int32(batches),
+		Metrics: &commonv1.Metrics{
+			AvgMetrics: makeMetrics(),
+		},
+	}
+
+	_, err := api.ReportTrialTrainingMetrics(ctx,
+		&apiv1.ReportTrialTrainingMetricsRequest{
+			TrainingMetrics: &trainingMetrics,
+		})
+	if err != nil {
+		return err
+	}
+
+	validationMetrics := trialv1.TrialMetrics{
+		TrialId:        trialID,
+		StepsCompleted: int32(batches),
+		Metrics: &commonv1.Metrics{
+			AvgMetrics: makeMetrics(),
+		},
+	}
+
+	_, err = api.ReportTrialValidationMetrics(ctx,
+		&apiv1.ReportTrialValidationMetricsRequest{
+			ValidationMetrics: &validationMetrics,
+		})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// PopulateExpTrialsMetrics adds metrics for a trial and exp to db.
+func PopulateExpTrialsMetrics(pgdb *db.PgDB, masterConfig *config.Config, trivialMetrics bool, batches int) error {
+	system := actor.NewSystem("mock")
+	ref, _ := system.ActorOf(sproto.AgentRMAddr, actor.ActorFunc(
+		func(context *actor.Context) error {
+			switch context.Message().(type) {
+			case sproto.DeleteJob:
+				context.Respond(sproto.EmptyDeleteJobResponse())
+			}
+			return nil
+		}))
+	mockRM := actorrm.Wrap(ref)
+	api := &apiServer{
+		m: &Master{
+			trialLogBackend: pgdb,
+			system:          system,
+			db:              pgdb,
+			taskLogBackend:  pgdb,
+			rm:              mockRM,
+			config:          masterConfig,
+			taskSpec:        &tasks.TaskSpec{},
+		},
+	}
+
+	resp, err := api.Login(context.TODO(), &apiv1.LoginRequest{Username: "admin"})
+	if err != nil {
+		return err
+	}
+
+	ctx := metadata.NewIncomingContext(context.TODO(),
+		metadata.Pairs("x-user-token", fmt.Sprintf("Bearer %s", resp.Token)))
+
+	// create exp and config
+	maxLength := expconf.NewLengthInBatches(100)
+	maxRestarts := 0
+	activeConfig := expconf.ExperimentConfig{ //nolint:exhaustivestruct
+		RawSearcher: &expconf.SearcherConfig{ //nolint:exhaustivestruct
+			RawMetric: ptrs.Ptr("loss"),
+			RawSingleConfig: &expconf.SingleConfig{ //nolint:exhaustivestruct
+				RawMaxLength: &maxLength,
+			},
+		},
+		RawEntrypoint:      &expconf.Entrypoint{RawEntrypoint: "model_def:SomeTrialClass"},
+		RawHyperparameters: expconf.Hyperparameters{},
+		RawCheckpointStorage: &expconf.CheckpointStorageConfig{ //nolint:exhaustivestruct
+			RawSharedFSConfig: &expconf.SharedFSConfig{ //nolint:exhaustivestruct
+				RawHostPath: ptrs.Ptr("/"),
+			},
+		},
+		RawMaxRestarts: &maxRestarts,
+	} //nolint:exhaustivestruct
+	activeConfig = schemas.WithDefaults(activeConfig)
+	model.DefaultTaskContainerDefaults().MergeIntoExpConfig(&activeConfig)
+
+	var defaultDeterminedUID model.UserID = 2
+	exp := &model.Experiment{
+		JobID:                model.NewJobID(),
+		State:                model.ActiveState,
```

Comment:
issue: I tried looking at the experiment in the webui and it doesn't show up. maybe because the experiment / trial are active? could we try creating them in "completed" state?